### ISSUE TITLE
How to solve this?

### ISSUE THREAD

**Issue by vkoptilov** [socketio/socket.io-client-swift#926]
![2018-01-08 22 04 55](https://user-images.githubusercontent.com/30573882/34689956-007420b8-f4c0-11e7-814c-c0741b5e7a65.png)

Please help before this works fine, but now I can't find solution.
When i try fix this issue with Xcode it doesn't help solve problem.

**Comment by nuclearace** [comment]
Read the [migration guide](https://nuclearace.github.io/Socket.IO-Client-Swift/12to13.html)

**Comment by vkoptilov** [comment]
NOT SOLVED!!!!

**Comment by nuclearace** [comment]
Try looking at the rest of the migration guide. It provides a more realistic starting point.

```swift
class Manager {
    let socketManager = SocketManager(socketURL: someURL)

    func addHandlers() {
        let socket = socketManager.socket(forNamespace: "/swift")

        // Add handlers
    }
}
```

**Comment by nuclearace** [comment]
The first part assumed that you were doing that in main. Those errors are just standard Swift errors telling you that you're doing it wrong. Instance member instantiation must either be done lazily or in an init method.

**Comment by vkoptilov** [comment]
![2018-01-08 23 10 29](https://user-images.githubusercontent.com/30573882/34692571-490ab1f8-f4c9-11e7-9cbe-c2369e6716fb.png)

this is my old code and early it works, but now i don't know how this should work^ please help

**Comment by nuclearace** [comment]
See: https://github.com/socketio/socket.io-client-swift/issues/926#issuecomment-356097180

Your getChannel can basically be

```swift
func getChannel() {
  let socket = manager.defaultSocket
  // All the rest of your code that references socket
}
```

**Comment by vkoptilov** [comment]
I don't understand anything what you write? My code was simple now I can't refactor it.
HELP ME, or how i can back SocketIO to previous normal working version

**Comment by vkoptilov** [comment]
![2018-01-08 23 22 16](https://user-images.githubusercontent.com/30573882/34693006-cb98b812-f4ca-11e7-989e-f1075e2e9654.png)

This is my old code and it work before^ now I don't understand what i need to do, help me refactor it

**Comment by vkoptilov** [comment]
I try read your documentation but it is not help me, I sped already 4 hours but it is not work

**Comment by nuclearace** [comment]
I can't write your code for you. All I can tell you is:

1. In v13 you create a `SocketManager` for a connection to socket.io.
2. The manager needs to be retained so that it isn't released by ARC, just like the clients used to be.
3. The manager has `socket(forNamespace:)` and `defaultSocket` for getting `SocketIOClient`s
4. Those SocketIOClients have almost the same API as they did in previous versions. Not much has changed. You can call `.on`, `.emit`, `.connect` like you did before.
5. You can call `socket(forNamespace:)` and `defaultSocket` and get back the same socket from the first call, so you don't need to store it anywhere, you just need the manager.

To sum up. You just need to add a manager property and replace all your old `socket.connect` etc with `manager.defaultSocket.connect` and `manager.socket.on`

**Comment by nuclearace** [comment]
I'm afraid I can't help anymore than that. If that's not enough, I suggest going back to the old version until you have the time to spend understanding the new API.

**Comment by vkoptilov** [comment]
THX, for help! Problem solved,
